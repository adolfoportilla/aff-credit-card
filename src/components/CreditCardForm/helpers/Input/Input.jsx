import React from "react";

import "./Input.scss";

const Input = ({
  autocomplete,
  baseClassName,
  inputMode,
  maxLength,
  name,
  onChange,
  placeholder,
  type,
  value,
  wrapperClassName
}) => {
  return (
    <div className={wrapperClassName}>
      <input
        type={type}
        name={name}
        onChange={onChange}
        placeholder={placeholder}
        className={baseClassName}
        maxLength={maxLength}
        value={value}
        inputMode={inputMode}
        autoComplete={autocomplete}
      />
    </div>
  );
};

export default Input;
