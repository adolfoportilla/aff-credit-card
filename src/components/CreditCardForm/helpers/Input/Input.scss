@import '../../../../scss/base.scss';

input {
  border: none;
  background: transparent !important;
  padding-left: 6px;
  width: 100%;

}

::placeholder {
  color: #bdc3c7;
}
