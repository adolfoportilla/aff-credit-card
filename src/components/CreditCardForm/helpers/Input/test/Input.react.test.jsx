import React from "react";
import { shallow } from "enzyme";

import Input from "../Input";

describe("Input", () => {
  const baseProps = {
    baseClassName: "base-class",
    value: "value"
  };

  test("renders basic layout", () => {
    const wrapper = shallow(<Input {...baseProps} />);

    expect(wrapper.find("input")).toHaveLength(1);
    expect(wrapper.find("input").props().value).toBe(baseProps.value);
    expect(wrapper.find("input").props().className).toBe(
      baseProps.baseClassName
    );
    expect(wrapper).toMatchSnapshot();
  });
});
