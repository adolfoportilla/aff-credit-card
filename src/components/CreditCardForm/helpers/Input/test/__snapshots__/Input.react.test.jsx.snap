// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders basic layout 1`] = `ShallowWrapper {}`;
